articles/why-and-how-zk-snark-works/3_non_interactive_zero-knowledge_of_a_polynomial.md

```diff
@@ -0,0 +1,259 @@
+
+## 3.1. Proving Knowledge of a Polynomial
+
+We initiate our exploration with the challenge of proving knowledge of a polynomial, gradually developing a generic approach and 
+uncovering various properties of polynomials along the way.
+
+Thus far, our discussions have centered on a rudimentary notion of proof, lacking mechanisms to enforce protocol rules. For instance, 
+the prover may not necessarily be acquainted with the polynomial and could employ alternative methods to yield a correct result. 
+Additionally, if the verifier's polynomial evaluations have a limited amplitude, such as 10, the verifier might guess a number with a 
+non-negligible probability of acceptance. Addressing these weaknesses becomes imperative, but first, let's delve into what it truly 
+means to "know" a polynomial.
+
+## 3.2. **Factorization**
+
+It states that any polynomial can be factored into linear polynomials as long as it is solvable. See the [lagrange interpolation](../../terms/lagrange_interpolation.md) for more informations.
+
+Example:
+$$(x - a_{0})(x - a_{1})...(x - a_{n}) = 0$$
+
+[Define](./notes.md) *p(x)* as a sample polynomial, and *p(x)* is the multiplication of those cofactors *t(x)*. So we have arbitrary
+polynomial:
+
+$$h(x) = \frac{p(x)}{t(x)}$$
+
+In particular:
+
+Given:
+$$p(x)  = x^3 - 3x^2 + 2x$$
+
+$$t(x) = (x-1)(x-2)$$
+
+- The Verifier samples a random value 23, calculates:
+ $$t = t(23) = (23 - 1)(23 - 2) = 462$$ And gives 23 to the Prover.
+
+- The Prover calculates $h(x) = \frac{p(x)}{t(x)}$ and evaluates
+ $$p = p(23) = 10626$$  
+ $$h = h(23) = 23$$ And provides $p, h$ to the Verifier.
+
+- The Verifier then checks that:  $$p = t \cdot h = 10626 = 462 \cdot 23$$
+
+   
+If this statement holds true, it implies that the prover provided an accurate polynomial.
+
+> [!Summary]
+> Prover may not know the claimed polynomial $p(x)$ at all.
+>
+> In the current protocol there is no enforcement of degree. Hence prover can cheat by using a polynomial of higher degree which also
+> satisfies the cofactors check.
+>
+> Because prover knows the random point $x = r$, he can construct any polynomial which has one shared point at $r$ with $t(r) · h(r)$.
+
+## 3.3 Obscure Evaluation
+
+Due to the information revealed in its raw form, the prover possesses knowledge of both the sample random value $r$ and its
+corresponding transformed value $t(r)$, which should remain concealed. Consequently, a mechanism akin to a hash function must be
+employed a process that, once executed, renders it challenging to reverse-engineer the original input.
+
+### 3.3.1 Homomorphic Encryption
+
+See the [Homomorphic Encryption](../../terms/homomorphic_encryption.md).
+
+### 3.3.2 Modular Arithmetic
+
+See the [Modular Arithmetic](../../terms/modular_arithmetic.md).
+
+### 3.3.3 Strong Homomorphic Encryption
+
+We use Modular Arithmetic for Homomorphic Encryption.
+
+### 3.3.4. Encrypted Polynomial
+
+Let's define $E$ :
+$$E(x) = g^{v} \pmod{n}$$
```

Comment:
I think this is E(v) instead of E(x)

Reply:
v is monomial so it has x in, you can see the example below